### PR TITLE
Open API

### PR DESCRIPTION
**Pull request by jcdickinson**
- Adding OpenAPI DTOs.

### CODE REVIEW

File: src/SourceCode.Clay.OpenApi/ApiKeySecurityScheme.cs [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+
+                hc = hc * 21 + base.GetHashCode();
+                if (Name != null) hc = hc * 21 + StringComparer.Ordinal.GetHashCode(Name);
+                hc = hc * 21 + Location.GetHashCode();
+
+                return ((int)(hc >> 32)) ^ (int)hc;
+            }
+        }
+
+        public static bool operator ==(ApiKeySecurityScheme scheme1, ApiKeySecurityScheme scheme2)
```
Comment (jcdickinson) [inline src/SourceCode.Clay.OpenApi/ApiKeySecurityScheme.cs]: NB: This equality operator honor inheritance.

File: src/SourceCode.Clay.OpenApi/ApiKeySecurityScheme.cs [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+        {
+            if (ReferenceEquals(scheme1, null) && ReferenceEquals(scheme2, null)) return true;
+            if (ReferenceEquals(scheme1, null) || ReferenceEquals(scheme2, null)) return false;
+            return scheme1.Equals((object)scheme2);
+        }
+
+        public static bool operator !=(ApiKeySecurityScheme scheme1, ApiKeySecurityScheme scheme2)
```
Comment (grant-d) [inline src/SourceCode.Clay.OpenApi/ApiKeySecurityScheme.cs]: Our existing conventions use (x, y)

File: src/SourceCode.Clay.OpenApi/Components.cs [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+using System.Collections.Generic;
+
+namespace SourceCode.Clay.OpenApi
+{
+    /// <summary>
+    /// Holds a set of reusable objects for different aspects of the OAS. All objects
+    /// defined within the components object will have no effect on the API unless they
+    /// are explicitly referenced from properties outside the components object.
+    /// </summary>
+    public class Components : IEquatable<Components>
```
Comment (grant-d) [inline src/SourceCode.Clay.OpenApi/Components.cs]: sealed?

**Comment by grant-d** [comment]
I am going to approve, but please fix small items in a separate PR

File: src/SourceCode.Clay.OpenApi/ApiKeySecurityScheme.cs (hunk 2 above)
Comment (jcdickinson) [reply src/SourceCode.Clay.OpenApi/ApiKeySecurityScheme.cs]: The equality refactoring has changed to use `foo1` and `foo2`. We should likely use that as it makes our lives easier in the long run.

File: src/SourceCode.Clay.OpenApi/Components.cs (hunk 3 above)
Comment (jcdickinson) [reply src/SourceCode.Clay.OpenApi/Components.cs]: Nope, nothing in the lib can be sealed - inheritance is how vendor extensions can be added.

**Pull request merged**